### DISCUSSION TITLE
how to run JS from htmxor

### DISCUSSION THREAD

**Discussion by eladmarg** [egil/Htmxor#35] [Q&A] [answered]
Hi,

I have a client side logic / libraries I would like to execute on every page.
(each page has it own logic / js)

it should be - 
1. execute on first time the page load.
2. execute after htmx replace the content.

what do you think its the best strategy to do this with this library?
there is samples for using tag htmx:afterRequest [here](https://stackoverflow.com/questions/73389202/how-to-execute-javascript-code-after-htmx-makes-an-ajax-request)

another option is just simply to include <script> tag in the response.

any ideas? what do you think?
thanks

**Comment by egil** [comment]
I'm a big fan of the declarative approach of both Blazor and htmx, thus, I think the following would be neat:

- for script that are site specific, these can be included like normal through App.razor as normal script tags.
- For page only scripts, they can be included on pages using a script tag. That tag could be included only when the page is requested (e.g., not when a request specifies a hx-target).
- for invoking a function when an element is added to the DOM, with the element as `this`, we could add a `hx-on:load="myFunction()"` attribute. It's not supported out of the box, but I don't see any downsides.

**Reply by egil** [reply]
Whoa, hang on, `hx-on::after-request` should work out of the box. Have you tried adding `<div hx-on::after-request="myFunction()">` and see if it calls `myFunctions` after the request?

**Comment by egil** [comment, marked as answer]
Here is an example how to do this:

https://github.com/egil/Htmxor/blob/main/samples/HtmxorExamples/Components/Pages/Examples/OutOfBandOutlets/ToastOutlet.razor
